### DISCUSSION TITLE
Random permissions error

### DISCUSSION THREAD

**Discussion by Myrmidonut** [pimcore/pimcore#14762] [Q&A / Support] [answered]
Hello

Ever since switching to Pimcore X, I am getting a seemingly random "permission denied" error message. It seems like the message shows up when I have been idle in Pimcore for some time, but I am not sure what is triggering it. It happens for admin or any other user. I keep seeing a call to "/admin/notification/find-last-unread" at the time when the message shows up, but I don't know if thats is the trigger or the result of the error.

I am running Apache with PHP-FPM, all set up through Plesk. It happens on different Pimcore X Versions with different PHP versions. I followed the documentation regarding file permissions for /var /public/var and the user. The PHP user has read and write access to everything. I cleared the cache and rebuild the symlinks via SSH with the same user.

Sometimes the error shows up when coming back to Pimcore after being idle for some time, sometimes it shows up right away when adding a redirect or any other modification. Sometimes it doesn't show up for days. After the error, any action in Pimcore keeps showing the error again, until you refresh the browser tab and you get redirected to /admin/login .

Am I missing a step in setting up a project? How can I go about debugging this? It seems like one of the admin JS files is running into this issue.

**usagelog.log shows this at the time of the error message:**

`2023-03-22T10:49:57+0100 : 2|Pimcore\Bundle\AdminBundle\Controller\Admin\NotificationController::findLastUnreadAction|pimcore_admin_notification_findlastunread|[]|{"lastUpdate":"1679478567"}

2023-03-22T10:50:27+0100 : 2|Pimcore\Bundle\AdminBundle\Controller\Admin\NotificationController::findLastUnreadAction|pimcore_admin_notification_findlastunread|[]|{"lastUpdate":"1679478597"}

2023-03-22T11:02:07+0100 : 0|Pimcore\Bundle\AdminBundle\Controller\Admin\LoginController::loginAction|pimcore_admin_login|[]|{"perspective":""}`

**The browser console shows this:**

`Failed to load resource: the server responded with a status of 403 (Forbidden)
http://www.example.com/admin/notification/find-last-unread?_dc=1679482472617&lastUpdate=1679482346

xhr request to /admin/notification/find-last-unread failed
http://www.example.com/bundles/pimcoreadmin/js/pimcore/startup.js?_dc=e47272e09c12046883f680e50571ab53491eac47`

**The error message in german:**

![Zugriff verweigert 11-55](https://user-images.githubusercontent.com/26467570/227933096-bf69b660-c7d6-4383-8fe2-1ccaf77ab344.png)

**Comment by podarcis** [comment]
The error message might be misleading as it's kind of generic for all kind of errors in communication. I suspect there is a problem with your PHP/Symfony **session handling/configuration**. Probably your session became invalid for some reason and Pimcore backend fails to communicate.

You may want to try a search for "Plesk Symfony session" to find related issues. Maybe it's a session-cleanup-script, maybe `session.gc_maxlifetime`, ...

By the way: You can easily reproduce this kind of error by modifying/invalidating your PHPSESSID cookie in your browser's dev-tools.

**Reply by Myrmidonut** [reply]
Thanks for pointing me into that direction, I will search for session related issues.

Removing the PHPSESSID cookie does indeed trigger the same result. I will keep an eye on what happens to that cookie when the error occurs next time.

**Comment by vmvartiainen** [comment]
@Myrmidonut did you find out what causes this?

I'm getting this same:

<img width="638" height="216" alt="image" src="https://github.com/user-attachments/assets/95dd3f62-b6bf-4503-8575-64e84df3f493" />

**Reply by Myrmidonut** [reply]
I never found out what actually causes this. It got better with Pimcore 11, I think. But that might not be true.

**Reply by kingjia90** [reply, marked as answer]
This is the message it appears when session is expired due to idle and user is logged out in another tab or so.

Maybe have a look and tweak your `gc_maxlifetime` settings (in the php.ini,htaccess and so), by default, it seems just 2 minutes
https://github.com/pimcore/admin-ui-classic-bundle/blob/10ea29130a41a9fefbcfc57b082f0ec5e3637a9b/src/Controller/Admin/IndexController.php#L283-L285

**Reply by vmvartiainen** [reply]
@kingjia90 I did this and it seems to work as well.

```
# config/packages/framework.yaml
framework:
    session:
        cookie_lifetime: 604800
        gc_maxlifetime: 604800
```
